### ISSUE TITLE
eslint linting inside temp loader folders

### ISSUE THREAD

**Issue by juancabrera** [eslint/eslint#8023]
**Tell us about your environment**

* **ESLint Version:** `3.14.1`
* **Node Version:**  `7.3.0`
* **npm Version:** `3.10.10`

**What parser (default, Babel-ESLint, etc.) are you using?**
`babel-loader` and `eslint-loader`

**Please show your full configuration:**

```shell
  module: {
    rules: [
      {
        enforce: 'pre', 
        test: /.vue$/,
        loader: 'eslint-loader',
        exclude: /\~/
      },
      {
        test: /\.vue$/,
        loader: 'vue-loader',
        options: {
          postcss: {
            plugins: [
              require('stylelint')(),
              require('postcss-import')(),
              require('postcss-cssnext')({
                browsers: 'last 2 versions'
              }),
              require("postcss-reporter")({ clearMessages: true }),
            ],
            options: {
              autoprefixer: false
            }
          }
        }
      },
      {
        test: /\.js$/,
        loaders: ["babel-loader", "eslint-loader"],
        include: projectRoot,
        exclude: /node_modules/,
      }
    ]
  }
```

But when I run webpack, got the linting warning and erros on the `.vue` files, which is cool, but also it run the lint in the temporary loader folders, here an example of the output I got:

```shell
WARNING in ./src/Application.vue

/Users/cabrera/htdocs/ssense-frontend/src/Application.vue
  10:43  warning  Extra semicolon  semi

✖ 1 problem (0 errors, 1 warning)

 @ ./src/client.js 2:0-36

WARNING in ./~/babel-loader/lib!./~/vue-loader/lib/selector.js?type=script&index=0!./~/eslint-loader!./src/Application.vue

/Users/cabrera/htdocs/ssense-frontend/src/Application.vue
  10:43  warning  Extra semicolon  semi

✖ 1 problem (0 errors, 1 warning)

 @ ./src/Application.vue 7:2-148
 @ ./src/client.js

WARNING in ./~/vue-loader/lib/template-compiler.js?id=data-v-65b562da!./~/vue-loader/lib/selector.js?type=template&index=0!./~/eslint-loader!./src/Application.vue

/Users/cabrera/htdocs/ssense-frontend/src/Application.vue
  10:43  warning  Extra semicolon  semi

✖ 1 problem (0 errors, 1 warning)

 @ ./src/Application.vue 9:2-207
 @ ./src/client.js

WARNING in ./~/css-loader!./~/vue-loader/lib/style-rewriter.js?id=data-v-65b562da!./~/vue-loader/lib/selector.js?type=styles&index=0!./~/eslint-loader!./src/Application.vue

/Users/cabrera/htdocs/ssense-frontend/src/Application.vue
  10:43  warning  Extra semicolon  semi

✖ 1 problem (0 errors, 1 warning)
```

I have tried to exclude the `/~/` temporary folder but is still linting inside that folder.

**Comment by alberto** [comment]
Try adding ~ to your .eslintignore file.

**Comment by juancabrera** [comment]
Hi @alberto, thanks for you response. I actually fixed this upgrading the version of `vue-loader` to `10.3.0`
